PlotSquared uses deprecated class MaterialData

### Server Implementation

Spigot

### Server Version

1.19.2

### Describe the bug

BlockEventListener line 940 calls BlockState#getData, which returns MaterialData. This ends up loading the Legacy Material Support even though the server is running 1.19.2 and using 1.19.2 blocks.
https://github.com/IntellectualSites/PlotSquared/blob/3a8fae47a00e6fa0bfdc50e93190c9f8b2606626/Bukkit/src/main/java/com/plotsquared/bukkit/listener/BlockEventListener.java#L940


### To Reproduce

1. Place a dispenser.
2. Put a water bucket in the dispenser.
3. Trigger the dispenser with your favorite redstone trigger.
4. See the _Initializing Legacy Material Support_ message in the console.

Note that the world doesn't even need to be a plot world! Bad PlotSquared, bad!

### Expected behaviour

PlotSquared doesn't load the Legacy Material Support on a 1.19.2 server using 1.19.2 blocks.

### Screenshots / Videos

_No response_

### Error log (if applicable)

_No response_

### Plot Debugpaste

Just look at your code.

### PlotSquared Version

6.10.4

### Checklist

- [X] I have included a Plot debugpaste.
- [X] I am using the newest build from https://www.spigotmc.org/resources/77506/ and the issue still persists.

### Anything else?

Problem also occurs using Paper.
```
2022-11-30 05:56:21 [ERROR] [STDERR] [org.bukkit.craftbukkit.v1_19_R1.legacy.CraftLegacy] Initializing Legacy Material Support. Unless you have legacy plugins and/or data this is a bug!
2022-11-30 05:56:21 [WARN] java.lang.Exception
2022-11-30 05:56:21 [WARN] 	at org.bukkit.craftbukkit.v1_19_R1.legacy.CraftLegacy.<clinit>(CraftLegacy.java:260)
2022-11-30 05:56:21 [WARN] 	at org.bukkit.craftbukkit.v1_19_R1.util.CraftMagicNumbers.getMaterial(CraftMagicNumbers.java:118)
2022-11-30 05:56:21 [WARN] 	at org.bukkit.craftbukkit.v1_19_R1.block.CraftBlockState.getData(CraftBlockState.java:154)
2022-11-30 05:56:21 [WARN] 	at PlotSquared.jar//com.plotsquared.bukkit.listener.BlockEventListener.onBlockDispense(BlockEventListener.java:940)
2022-11-30 05:56:21 [WARN] 	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor71.execute(Unknown Source)
2022-11-30 05:56:21 [WARN] 	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77)
2022-11-30 05:56:21 [WARN] 	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
2022-11-30 05:56:21 [WARN] 	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
2022-11-30 05:56:21 [WARN] 	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:670)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.core.dispenser.IDispenseBehavior$9.a(IDispenseBehavior.java:692)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.core.dispenser.DispenseBehaviorItem.dispense(DispenseBehaviorItem.java:32)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.world.level.block.BlockDispenser.a(BlockDispenser.java:98)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.world.level.block.BlockDispenser.a(BlockDispenser.java:124)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.world.level.block.state.BlockBase$BlockData.a(BlockBase.java:1014)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.server.level.WorldServer.d(WorldServer.java:1141)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.world.ticks.TickListServer.a(TickListServer.java:197)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.world.ticks.TickListServer.a(TickListServer.java:94)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.server.level.WorldServer.a(WorldServer.java:673)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1535)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:446)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1397)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.server.MinecraftServer.v(MinecraftServer.java:1173)
2022-11-30 05:56:21 [WARN] 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:305)
2022-11-30 05:56:21 [WARN] 	at java.base/java.lang.Thread.run(Thread.java:833)
```

--- a/Bukkit/src/main/java/com/plotsquared/bukkit/listener/BlockEventListener.java
+++ b/Bukkit/src/main/java/com/plotsquared/bukkit/listener/BlockEventListener.java
@@ -101,7 +101,6 @@ import org.bukkit.event.block.EntityBlockFormEvent;
 import org.bukkit.event.block.LeavesDecayEvent;
 import org.bukkit.event.block.SpongeAbsorbEvent;
 import org.bukkit.event.world.StructureGrowEvent;
-import org.bukkit.material.Directional;
 import org.bukkit.projectiles.BlockProjectileSource;
 import org.bukkit.util.Vector;
 import org.checkerframework.checker.nullness.qual.NonNull;

--- a/Bukkit/src/main/java/com/plotsquared/bukkit/listener/BlockEventListener.java
+++ b/Bukkit/src/main/java/com/plotsquared/bukkit/listener/BlockEventListener.java
@@ -71,6 +71,7 @@ import org.bukkit.block.Block;
 import org.bukkit.block.BlockFace;
 import org.bukkit.block.BlockState;
 import org.bukkit.block.data.BlockData;
+import org.bukkit.block.data.type.Dispenser;
 import org.bukkit.entity.Entity;
 import org.bukkit.entity.Fireball;
 import org.bukkit.entity.Player;
@@ -926,6 +927,9 @@ public class BlockEventListener implements Listener {
 
     @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
     public void onBlockDispense(BlockDispenseEvent event) {
+        if (!this.plotAreaManager.hasPlotArea(event.getBlock().getWorld().getName())) {
+            return;
+        }
         Material type = event.getItem().getType();
         switch (type.toString()) {
             case "SHULKER_BOX", "WHITE_SHULKER_BOX", "ORANGE_SHULKER_BOX", "MAGENTA_SHULKER_BOX", "LIGHT_BLUE_SHULKER_BOX",
@@ -937,7 +941,7 @@ public class BlockEventListener implements Listener {
                 if (event.getBlock().getType() == Material.DROPPER) {
                     return;
                 }
-                BlockFace targetFace = ((Directional) event.getBlock().getState().getData()).getFacing();
+                BlockFace targetFace = ((Dispenser) event.getBlock().getBlockData()).getFacing();
                 Location location = BukkitUtil.adapt(event.getBlock().getRelative(targetFace).getLocation());
                 if (location.isPlotRoad()) {
                     event.setCancelled(true);

